mining/src/mempool/validate_and_insert_transaction.rs

```diff
@@ -73,9 +74,30 @@ impl Mempool {
         self.validate_transaction_in_context(&transaction)?;
 
         // Before adding the transaction, check if there is room in the pool
-        self.transaction_pool.limit_transaction_count(1, &transaction)?.iter().try_for_each(|x| {
-            self.remove_transaction(x, true, TxRemovalReason::MakingRoom, format!(" for {}", transaction_id).as_str())
-        })?;
+        let transaction_size = transaction.tx.estimate_mem_bytes();
+        let txs_to_remove = self.transaction_pool.limit_transaction_count(&transaction)?;
+        for x in txs_to_remove.iter() {
+            self.remove_transaction(x, true, TxRemovalReason::MakingRoom, format!(" for {}", transaction_id).as_str())?;
+            // self.transaction_pool.limit_transaction_count(&transaction) returns the
+            // smallest prefix of `ready_transactions` (sorted by ascending fee-rate)
+            // that makes enough room for `transaction`, but since each call to `self.remove_transaction`
+            // also removes all transactions dependant on `x` we might already have enough room.
+            if self.transaction_pool.len() < self.config.maximum_transaction_count
```

Comment:
Same question here. Shouldn't this be `self.transaction_pool.len() <= self.config.maximum_transaction_count` ?